packages/wallet-sdk/src/provider/MobileRelayUI.ts

```diff
@@ -0,0 +1,169 @@
+import { RedirectDialog } from "../components/RedirectDialog/RedirectDialog";
+import { ErrorHandler } from "../errors";
+import {
+  EthereumAddressFromSignedMessageRequest,
+  SignEthereumMessageRequest,
+  SignEthereumTransactionRequest,
+  SubmitEthereumTransactionRequest,
+} from "../relay/Web3Request";
+import {
+  EthereumAddressFromSignedMessageResponse,
+  SignEthereumMessageResponse,
+  SignEthereumTransactionResponse,
+  SubmitEthereumTransactionResponse,
+} from "../relay/Web3Response";
+import { AddressString, ProviderType } from "../types";
+import { WalletUI } from "./WalletUI";
+
+// TODO: Implement & present in-page wallet picker instead of navigating to www.coinbase.com/connect-dapp
+export class MobileRelayUI implements WalletUI {
+  private readonly redirectDialog: RedirectDialog;
+  private attached = false;
+
+  constructor() {
+    this.redirectDialog = new RedirectDialog();
+  }
+
+  attach() {
+    if (this.attached) {
+      throw new Error("Coinbase Wallet SDK UI is already attached");
```

Comment:
should we actually throw here or is it better to just return early

Reply:
it followed the same pattern as the existing class 
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/35c32729-ef06-43ea-b648-df7bad0a2934)

Reply:
cool that works then